packages/input-selection/src/types.ts

```diff
@@ -35,12 +35,17 @@ export interface SelectionResult {
    * has removed values to pay for entries in the requested output set.
    */
   remainingUTxO: Set<Cardano.Utxo>;
+
+  /** The list of redeemers and their execution cost. */
+  redeemers?: Array<Cardano.Redeemer>;
 }
 
 /**
  * @returns minimum transaction fee in Lovelace.
  */
-export type EstimateTxFee = (selectionSkeleton: SelectionSkeleton) => Promise<Cardano.Lovelace>;
+export type EstimateTxFee = (
+  selectionSkeleton: SelectionSkeleton
+) => Promise<{ fee: bigint; redeemers?: Array<Cardano.Redeemer> }>;
```

Comment:
`fee` has a very specific meaning, maybe this?
```suggestion
export type TxCosts = { fee: bigint; redeemers?: Array<Cardano.Redeemer> };
export type EstimateTxCosts = (selectionSkeleton: SelectionSkeleton) => Promise<TxCosts>;
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1299/commits/40ee3afdb0010b2cd8e9de831639775c0dd3873a